### PR TITLE
Add subscription property to SubscriptionItem for compatibility

### LINKED ISSUE #493: Error using "Create Resource..." from subscription item context menu
image

### PR DESCRIPTION
**Pull request by alexweininger**
Fixes #493

### CODE REVIEW

File: src/tree/v2/azure/SubscriptionItem.ts [hunk 1]
```diff
@@ -25,13 +26,16 @@ export class SubscriptionItem implements ResourceGroupsItem {
...
         private readonly resourceProviderManager: AzureResourceProviderManager,
-        public readonly subscription: AzureSubscription) {
+        public readonly v2Subscription: AzureSubscription) {
     }
 
-    public readonly id: string = `/subscriptions/${this.subscription.subscriptionId}`;
+    public readonly id: string = `/subscriptions/${this.v2Subscription.subscriptionId}`;
+
+    // for v1.5 compatibility
+    public readonly subscription = createSubscriptionContext(this.v2Subscription);
```
[the remark this answers is not in the document]
Comment (alexweininger) [reply src/tree/v2/azure/SubscriptionItem.ts:35]: Definitely worth a shot. I wasn't looking forward to having to use `v2Subscription` later, and then try to change it back to `subscription` once the compat stuff is gone.

[the remark this answers is not in the document]
Comment (alexweininger) [reply src/tree/v2/azure/SubscriptionItem.ts]: Works!

File: src/tree/v2/azure/SubscriptionItem.ts [hunk 2]
```diff
@@ -25,10 +26,19 @@ export class SubscriptionItem implements ResourceGroupsItem {
         private readonly context: ResourceGroupsTreeContext,
         private readonly resourceGroupingManager: AzureResourceGroupingManager,
         private readonly resourceProviderManager: AzureResourceProviderManager,
-        public readonly subscription: AzureSubscription) {
+        subscription: AzureSubscription) {
+
+        this.subscription = {
+            // for v1.5 compatibility
+            ...createSubscriptionContext(subscription),
```
Comment (bwateratmsft) [inline src/tree/v2/azure/SubscriptionItem.ts:33]: Are there any property collisions?

Comment (alexweininger) [reply src/tree/v2/azure/SubscriptionItem.ts:33]: The types have overlapping properties, but they all have the same types. So it should be fine.

**Review by bwateratmsft — APPROVED** [review]

**Pull request merged**